server/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -448,12 +609,20 @@ private XContentBuilder toXContentSnapshot(final XContentBuilder builder, final
         return builder;
     }
 
+    public static SnapshotInfo fromXContent(final XContentParser parser) throws IOException {
+        parser.nextToken();
```

Comment:
maybe validate that these tokens are thee right start_object or whatever ?

Reply:
Done.